Komtur behaviour

d-schn opened the pull request:
Fixed a bug where komtur would always go into chase after cooldown and Christine outside of chase area

Komtur will try to intercept Christine:
- walk towards her if she is less than INTERCEPT_RADIUS from castle
- walk towards the closest point on INTERCEPT_RADIUS from Christine
Komtur will not chase Christine too far away from castle

closes #80

d-schn committed:
Fix komtur cooldown to chase

Fixed a bug where komtur would always go into chase after cooldown and christine outside of chase area

d-schn committed:
behaviour descr

specified behaviour in comment

d-schn committed:
Komtur interception

Komtur will try to intercept Christine:
- walk towards her if she is less than INTERCEPT_RADIUS from castle
- walk towards the closest point on INTERCEPT_RADIUS from Christine
Komtur will not chase Christine too far away from castle

d-schn committed:
Merge branch 'main' into komtur_behaviour

g-oelk reviewed (changes requested):
komtur is still rather aggressive but easier to get him out of chasing mode. Time loss penalty should be removed and later replaced by slow aura #90

d-schn committed:
time malus removed and slower komtur

Komtur no longer steals a day from Christine, is slightly slower, thus hopefully less aggressive

b-irsigler commented:
Personally, I find him still too aggressive and stressful. Let's discuss all together.

There was ides that the Komtur could have another state - inside the castle.

b-irsigler committed:
Reduced Komtur motion speed even further

b-irsigler reviewed (approved):
There is open discussion about the aggression of the Komtur. However, will merge this now as the current state is working. Optimization can be tackled in #92
